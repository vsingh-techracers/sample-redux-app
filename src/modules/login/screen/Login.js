/* eslint class-methods-use-this: 0 */
import React, { Component, PropTypes } from 'react';
import {
  Text,
  View,
  TouchableOpacity,
  TextInput,
  StatusBar,
  Alert,
  ActivityIndicator,
  Keyboard,
} from 'react-native';
import { connect } from 'react-redux';
import _ from 'lodash';
import loginActions from 'login/actions';
// import { BaseStyle } from 'src/styles';
import { styles as loginStyles } from '../styles/LoginStyles';

console.ignoredYellowBox = ['Warning: View.propTypes'];

export class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: 'xyz@abc.com',
      password: '123456789',
      isLoginSuccess: false,
    };
  }

  componentWillMount() {
    Keyboard.dismiss();
  }

  componentWillReceiveProps(nextProps) {
    if (!nextProps.isLoginSuccess && !this.state.isLoginSuccess && !nextProps.isLoading) {
      this.setState({ isLoginSuccess: true });
      Alert.alert(
      'Error!',
      'There was a problem in handling your process. Please try again later',
        [
          { text: 'OK' },
        ],
        { cancelable: false },
      );
    }
  }

  clearText() {
    this.setState({
      email: '',
    });
  }

  loginRequest() {
    this.setState({ isLoginSuccess: false });
    this.props.loginRequest(this.state.email, this.state.password);
  }

  render() {
    const activeText = this.state.email === '' ? 0.5 : 1;
    return (
      <View style={loginStyles.container}>
        <StatusBar
          barStyle="light-content"
        />
        <View style={loginStyles.container}>
          <View style={loginStyles.viewFlex}>
            <View style={loginStyles.viewNextButton}>
              <TouchableOpacity
                style={loginStyles.buttonNext}
                onPress={() => this.loginRequest()}
              >
                <Text style={[loginStyles.buttonText, { opacity: activeText }]}>{'Next'}</Text>
              </TouchableOpacity>
            </View>
          </View>
          <View style={loginStyles.viewFlexThree}>
            <View style={loginStyles.viewFlexThree}>
              <View style={loginStyles.containView}>
                <View style={loginStyles.placeHolder}>
                  <Text style={loginStyles.emailStyle}>E-mail Address</Text>
                </View>
                <View style={loginStyles.bottomView}>
                  <View style={loginStyles.viewFlexNine}>
                    <TextInput
                      style={loginStyles.textInput}
                      onChangeText={email => this.setState({ email })}
                      placeholder={'Email Address'}
                      keyboardType={'email-address'}
                      placeholderTextColor={'#7F7F7F'}
                      autoFocus={Boolean(true)}
                      value={this.state.email}
                    />
                  </View>
                  {
                    this.state.email ?
                      <View style={loginStyles.viewFlex}>
                        <TouchableOpacity
                          style={loginStyles.clearButton}
                          onPress={() => this.clearText()}
                        >
                          <Text style={loginStyles.buttonText}>{'X'}</Text>
                        </TouchableOpacity>
                      </View>
                    : <View style={loginStyles.viewFlex} />
                  }
                </View>
                {
                  this.props.isLoading ?
                    <ActivityIndicator
                      color={'#bc2b78'}
                      size={'large'}
                      style={loginStyles.activityIndicator}
                    /> : null
                }
              </View>
            </View>
          </View>
        </View>
      </View>
    );
  }
}

Login.propTypes = {
  loginRequest: PropTypes.func,
  isLoading: PropTypes.bool,
  isLoginSuccess: PropTypes.bool,
  loginResponse: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.any),
    PropTypes.objectOf(PropTypes.any),
  ]),
};

Login.defaultProps = {
  loginRequest: _.noop,
  loginResponse: [],
  isLoading: false,
  isLoginSuccess: false,
};

const mapStateToProps = state => ({
  loginResponse: state.loginReducer.loginResponse,
  isLoading: state.loginReducer.isLoading,
  isLoginSuccess: state.loginReducer.isLoginSuccess,
});

const mapDispatchToProps = () => loginActions;

const loginScreen = connect(mapStateToProps, mapDispatchToProps)(Login);

export default loginScreen;

