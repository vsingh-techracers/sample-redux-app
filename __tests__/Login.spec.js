// __tests__/Login-test.js // this is another file convention

import 'react-native';
import React from 'react';
// import mockStore from 'redux-mock-store';
import { shallow, render } from 'enzyme';
import { Login } from '../src/modules/login/screen/Login';

describe('Testing Login component', () => {
  it('Shallow Rendering', () => {
    const wrapper = shallow(
      <Login isFromForgotPassword={Boolean(true)} />,
    );

    // console.log('Shallow Rendering => ', wrapper.instance());
    console.log('Shallow Rendering 2 => ', wrapper.find('clearText').dive());

    console.log('Shallow Rendering => props =>', wrapper.instance().props);
    console.log('Shallow Rendering => state =>', wrapper.instance().state);
    console.log('Shallow Rendering => props => isLoading', wrapper.instance().props.isLoading);
    console.log('Shallow Rendering => props => email', wrapper.instance().state.email);

    /* create snapshot */
    expect(wrapper).toMatchSnapshot();

    /* Change and test value of props */
    wrapper.setProps({ isLoading: false });
    expect(wrapper.instance().props.isLoading).toEqual(false);

    /* Change and test value of state */
    wrapper.instance().setState({ email: 'abc.com' });
    expect(wrapper.state('email')).toBe('abc.com');
  });

  // it('Full Rendering', () => {
  //   const wrapper = mount(
  //     <Login />,
  //   );
  //   console.log('Full Rendering => ', wrapper);
  //   expect(wrapper).toMatchSnapshot();
  // });

  it('Static Rendering', () => {
    const wrapper = render(
      <Login isFromForgotPassword={Boolean(true)} />,
    );
    console.log('Static Rendering => ', wrapper);
    expect(wrapper).toMatchSnapshot();
  });
});

/* using import mockStore from 'redux-mock-store'; */

// const initialState = {
//   lookupEmailResponse: {
//     status: 200,
//   },
//   isLoading: false,
//   isLoginSuccess: true,
// };

// describe('Testing Login component', () => {
//   it('renders as expected', () => {
//     const wrapper = shallow(
//       <Login />,
//       { context: { store: mockStore(initialState) } },
//     );
//     expect(wrapper.dive()).toMatchSnapshot();
//   });
// });

// describe('Testing Login component', () => {
//   it('Testing Login component updates', () => {
//   const component = mount(
//       <Login />
//     );
//     // console.log(wrapper.instance().props);
//     // expect(wrapper).toMatchSnapshot();
//     // wrapper.setProps({ isLoading: false });
//     // console.log(wrapper.instance().props);
//     // expect(wrapper).toMatchSnapshot();
//     // wrapper.setProps({ isFromForgotPassword: false });
//     // console.log(wrapper.instance().props);
//     // expect(wrapper).toMatchSnapshot();
//     //wrapper.setProps({ isFromForgotPassword: true });
//     console.log('==========', component);
//     //expect(wrapper).toMatchSnapshot();
//   });
// });


// it('Shallow Rendering', () => {
//     const wrapper = shallow(
//       <Login isFromForgotPassword={true} />
//     );
//     console.log(wrapper.instance().props);
//     // expect(wrapper).toMatchSnapshot();
//     wrapper.setProps({ isLoading: false });
//     console.log('============2323', wrapper.instance().props.isLoading);
//     expect(wrapper.instance().props.isLoading).toEqual(false);
//     // console.log(wrapper.instance().props);
//     // expect(wrapper).toMatchSnapshot();
//     // wrapper.setProps({ isFromForgotPassword: false });
//     // console.log(wrapper.instance().props);
//     // expect(wrapper).toMatchSnapshot();
//     //wrapper.setProps({ isFromForgotPassword: true });
//     console.log('==========', wrapper.instance().props);
//     //expect(wrapper).toMatchSnapshot();
//   });



