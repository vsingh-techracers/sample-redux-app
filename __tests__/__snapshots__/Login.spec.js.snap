// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login component Shallow Rendering 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(226, 80, 87)",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <StatusBar
    animated={false}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <View
    style={
      Object {
        "backgroundColor": "rgb(226, 80, 87)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "height": 45,
            "justifyContent": "flex-end",
            "marginTop": 20,
            "width": 750,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Object {
              "height": 45,
              "justifyContent": "flex-end",
              "width": 65,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "800",
                  "textAlign": "center",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            Next
          </Text>
        </TouchableOpacity>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <View
          style={
            Object {
              "height": 100,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontSize": 14,
                  "fontWeight": "800",
                  "padding": 10,
                }
              }
            >
              E-mail Address
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 9,
                }
              }
            >
              <TextInput
                autoFocus={true}
                keyboardType="email-address"
                onChangeText={[Function]}
                placeholder="Email Address"
                placeholderTextColor="#7F7F7F"
                style={
                  Object {
                    "color": "white",
                    "flex": 1,
                    "fontSize": 22,
                    "fontWeight": "300",
                    "height": 40,
                    "padding": 10,
                  }
                }
                value="xyz@abc.com"
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                focusedOpacity={0.7}
                onPress={[Function]}
                style={
                  Object {
                    "height": 40,
                    "justifyContent": "center",
                    "marginTop": 5,
                    "width": 30,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 16,
                      "fontWeight": "800",
                      "textAlign": "center",
                    }
                  }
                >
                  X
                </Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Testing Login component Shallow Rendering 2`] = `
<View
  style={
    Object {
      "backgroundColor": "rgb(226, 80, 87)",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "width": 750,
    }
  }
>
  <StatusBar
    animated={false}
    barStyle="light-content"
    showHideTransition="fade"
  />
  <View
    style={
      Object {
        "backgroundColor": "rgb(226, 80, 87)",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "height": 45,
            "justifyContent": "flex-end",
            "marginTop": 20,
            "width": 750,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          focusedOpacity={0.7}
          onPress={[Function]}
          style={
            Object {
              "height": 45,
              "justifyContent": "flex-end",
              "width": 65,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 16,
                  "fontWeight": "800",
                  "textAlign": "center",
                },
                Object {
                  "opacity": 1,
                },
              ]
            }
          >
            Next
          </Text>
        </TouchableOpacity>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 3,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 3,
          }
        }
      >
        <View
          style={
            Object {
              "height": 100,
            }
          }
        >
          <View
            style={
              Object {
                "height": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": "white",
                  "fontSize": 14,
                  "fontWeight": "800",
                  "padding": 10,
                }
              }
            >
              E-mail Address
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "height": 50,
                "justifyContent": "center",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 9,
                }
              }
            >
              <TextInput
                autoFocus={true}
                keyboardType="email-address"
                onChangeText={[Function]}
                placeholder="Email Address"
                placeholderTextColor="#7F7F7F"
                style={
                  Object {
                    "color": "white",
                    "flex": 1,
                    "fontSize": 22,
                    "fontWeight": "300",
                    "height": 40,
                    "padding": 10,
                  }
                }
                value="xyz@abc.com"
              />
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                }
              }
            >
              <TouchableOpacity
                activeOpacity={0.2}
                focusedOpacity={0.7}
                onPress={[Function]}
                style={
                  Object {
                    "height": 40,
                    "justifyContent": "center",
                    "marginTop": 5,
                    "width": 30,
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": "#FFFFFF",
                      "fontSize": 16,
                      "fontWeight": "800",
                      "textAlign": "center",
                    }
                  }
                >
                  X
                </Text>
              </TouchableOpacity>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Testing Login component Static Rendering 1`] = `
<view
  style="flex:1;flex-direction:column;justify-content:center;background-color:rgb(226, 80, 87);width:750px;"
>
  <view
    style="flex:1;flex-direction:column;justify-content:center;background-color:rgb(226, 80, 87);width:750px;"
  >
    <view
      style="flex:1;flex-direction:row;"
    >
      <view
        style="width:750px;height:45px;justify-content:flex-end;align-items:flex-end;margin-top:20px;"
      >
        <view
          style="width:65px;height:45px;justify-content:flex-end;opacity:1;"
        >
          <text
            style="0:[object Object];1:[object Object];"
          >
            Next
          </text>
        </view>
      </view>
    </view>
    <view
      style="flex:3;"
    >
      <view
        style="flex:3;"
      >
        <view
          style="height:100px;"
        >
          <view
            style="height:30px;"
          >
            <text
              style="font-size:14px;font-weight:800;padding:10px;color:white;"
            >
              E-mail Address
            </text>
          </view>
          <view
            style="height:50px;flex-direction:row;justify-content:center;"
          >
            <view
              style="flex:9;"
            >
              <textinput
                placeholder="Email Address"
                style="flex:1;height:40px;padding:10px;font-size:22px;color:white;font-weight:300;"
                value="xyz@abc.com"
              />
            </view>
            <view
              style="flex:1;flex-direction:row;"
            >
              <view
                style="width:30px;height:40px;justify-content:center;margin-top:5px;opacity:1;"
              >
                <text
                  style="color:#FFFFFF;font-size:16px;font-weight:800;text-align:center;"
                >
                  X
                </text>
              </view>
            </view>
          </view>
        </view>
      </view>
    </view>
  </view>
</view>
`;
